### DISCUSSION TITLE
Runners AMI change causes user_data to be changed...

### DISCUSSION THREAD

**Discussion by anthonydahanne** [npalm/terraform-aws-gitlab-runner#462] [Q&A]
Hello !
So I was just wondering if there was a way to avoid triggering changes to a `plan` / `apply` when the `runners_ami`, fetched via datasource + filter on Amazon EC2 AMI, is updated.

Since this [template](https://github.com/npalm/terraform-aws-gitlab-runner/blob/develop/template/runner-config.tpl#L62) requires it, the `user_data` of the runners will be updated whenever I re run plan / apply after AWS updated the AMI.

I know it's kind of "by design" - but I wish there was a way to tell AWS "I want the latest AMI" without having to use TF to hardcode the latest value in the `user_data` - not mentioning that the `user_data` diff is encoded in base64, so it's not super convenient to find out such changes...

Please let me know if my question makes sense, 
Thank you!

**Comment by kayman-mk** [comment]
As long as we want to use the most recent AMI, this is impossible to achieve. There is no such option available.

We could calculate the AMI at runtime. But the drawback is that the `terraform apply` is no longer idempotent as the AMI might change. So it wouldn't be a good option.

**Reply by anthonydahanne** [reply]
yeah, that makes sense. oh well, I'll live with those AMI changes!
